packages/batch/src/SqsFifoPartialProcessor.ts

```diff
@@ -94,16 +153,44 @@ class SqsFifoPartialProcessor extends BatchProcessorSync {
     const remainingRecords = this.records.slice(firstFailureIndex);
 
     for (const record of remainingRecords) {
-      const data = this.toBatchType(record, this.eventType);
-      processedRecords.push(
-        this.failureHandler(data, new SqsFifoShortCircuitError())
-      );
+      this.processFailRecord(record, new SqsFifoShortCircuitError());
     }
 
     this.clean();
 
     return processedRecords;
   }
+
+  /**
+   * Adds the specified group ID to the set of failed group IDs.
+   * @param group - The group ID to be added to the set of failed group IDs.
+   */
+  private addToFailedGroup(group: string): void {
+    this.failedGroupIds.add(group);
+  }
+
+  /**
+   * Processes a fail record.
+   * @param record - The record that failed.
+   * @param exception - The error that occurred.
+   * @returns The failure response.
```

Comment:
In the docstrings, let's:
- leave 1 space/new-line between description and params
- remove the `@returns` 

for example

```ts
/**
  * Processes a fail record.
  *
  * @param record - The record that failed.
  * @param exception - The error that occurred.
  */
```